### PR TITLE
Convert Chooser from Vue.js SPA to Nuxt (static)

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #279 by @obulat

## Description

This PR is an experiment of converting the Chooser from a Vue SPA app to Nuxt in an attempt to improve SEO and performance.

Nuxt enables us to create a static site, which means that even a user with Javascript disabled can see some information on the site, even though they won't be able to interact with it. This is also good for SEO for search engines other than Google that cannot render JS SPA apps.

This PR improves the SEO and possibly UX for slower connections by creating a statically rendered index page that is initially sent to the user instead of the current generic "You cannot view this page without JS" provided by VueJS.

However, it does not deliver on making the bundle size smaller. The lighthouse score, after adding in the Sentry, Hotjar and Google Analytics integrations becomes even worse than the current VueJS score (the one in the #311 PR).

I also couldn't set up the PurgeCSS to work correctly. It either removes the basic CSS that is declared in the Vocabulary / Bulma `generic.sass` file, removing the body default font-family of Source Sans Pro, or leaves too much of unused CSS in the final build.

If we can solve the problem of bundle size and CSS size, I believe Nuxt could be better for the Chooser for one more reason:
Nuxt makes it easier to add a PWA to the app, which might be useful for the users of the Chooser on unstable connections or on mobiles.

## Checklist
 
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Cronus1007** [comment]
@obulat Can I also be a part of this chane from vue to nuxt since I have previously worked with `nuxt`.Would love to make this transition.

**Comment by obulat** [comment]
> @obulat Can I also be a part of this change from vue to Nuxt since I have previously worked with `nuxt`.Would love to make this transition.

I have updated the comment, so you can get some more context on the change. If you have any solutions for the problems mentioned or any improvements, please feel free to join the discussion.

**Comment by Cronus1007** [comment]
@obulat Would like to start while updating the `README` file with the required changes and keep it upto date.

**Comment by Cronus1007** [comment]
@obulat While running `npm test` all the test cases are passing but leaving behind few console `errors and warnings`.And `npm test` only runs the unit test cases rather than both unit and e2e.

**Comment by rohan-goel** [comment]
@obulat If I am not wrong vue uses webpack under the hood and we can optimize its configuration. Usually a webpack.config.js file is used to configure webpack but with vue we need to add those configuration inside vue.config.js.

**Review by zackkrida — CHANGES_REQUESTED** [review]
Hi @obulat, thanks for this PR and this work. It was extremely helpful to be able to compare these two projects side by side. And whether we use Nuxt or not, there's lots of improvements to config and docs in here that I'd like to see moved over.

There's a lot of things I like about adding nuxt, but there's also some tradeoffs. At the end of the day, it leads us to send more JavaScript to the user than we would without it. 

The current site gets its first initial paint (which is also it's first _contentful_ paint, which is good news) in 4.732s on a good 3g connection. This is actually pretty good for a client-side JavaScript site like this. The site sends about 724kb or resources, but goes over 1mb with Nuxt implemented.

Basically, if you think the benefits of nuxt to developer experience, code quality and organization, and our ability to optimize performance in the future (perhaps with built-in performance features of nuxt) are worth the initial performance being made _worse_, I'd be open to merging this. But if the goal is to improve performance, I don't think Nuxt is the right fit.

**Comment by obulat** [comment]
Thank you for your detailed review, @zackkrida!

I believe that we should prioritize improvement in performance for the end-users because we want users on any connection to be able to use the Chooser. Over 1mb of resources for a site almost without images in this PR is too much... Granted, the large portion of the resources is the third-party code (Hotjar, Sentry and Google Analytics), we still seem to send too much code for slow connection. As someone using slow mobile network at the moment, I can feel the pain.

As for the improvements to code quality and organization, I will create issues for specific improvements to add to the main branch.
